### PR TITLE
obj-265 ensure values are kept on error page

### PR DESCRIPTION
**Pull request by mattch1**
Added values to name page, so that they are held when an error is displayed.

### CODE REVIEW

File: src/controllers/objecting.entity.name.controller.ts [hunk 1]
```diff
@@ -57,7 +57,12 @@ const showErrorsOnScreen = (errors: Result, res: Response) => {
       }
       errorListData.push(govUkErrorData);
     });
+
+  const fullNameValue: string = req.body.fullName
   return res.render(Templates.OBJECTING_ENTITY_NAME, {
+    fullNameValue,
```
Comment (markpit) [inline src/controllers/objecting.entity.name.controller.ts:63]: can we add a unit test to cover this new code?

File: views/objecting-entity-name.html [hunk 2]
```diff
@@ -60,11 +61,13 @@
...
               value: "yes",
-              text: "Yes"
+              text: "Yes",
+              checked: isYesChecked
             },
             {
               value: "no",
-              text: "No"
+              text: "No",
+              checked: isNoChecked
```
Comment (ICoxCH) [inline views/objecting-entity-name.html:70]: Could this be simplified to one boolean?
isChecked, and this value be !isChecked

**Comment by SVarallo** [comment]
This should be for story: https://companieshouse.atlassian.net/browse/OBJ-267 not OBJ-265. PR should be renamed to show this.

File: views/objecting-entity-name.html (hunk 2 above)
Comment (mattch1) [reply views/objecting-entity-name.html:70]: I think these need to be dealt with separately so as to cover the 3 senarios:
Neither selected
Yes selected
No selected.

File: src/controllers/objecting.entity.name.controller.ts (hunk 1 above)
Comment (mattch1) [reply src/controllers/objecting.entity.name.controller.ts:63]: Unit test coverage added.

File: views/objecting-entity-name.html (hunk 2 above)
Comment (ICoxCH) [reply views/objecting-entity-name.html:70]: I forgot about the Neither selected scenario, fair enough

File: src/controllers/objecting.entity.name.controller.ts (hunk 1 above)
Comment (SVarallo) [reply src/controllers/objecting.entity.name.controller.ts:63]: Will need a unit test to cover the retention of the name as well

Comment (SVarallo) [reply src/controllers/objecting.entity.name.controller.ts:63]: Scratch that, it's covered

File: test/controller/objecting.entity.name.controller.spec.unit.ts [hunk 3]
```diff
@@ -88,6 +88,22 @@ describe("objecting entity name tests", () => {
     expect(response.status).toEqual(200);
     expect(response.text).toContain(ENTER_FULL_NAME);
     expect(response.text).not.toContain(SELECT_TO_DIVULGE);
+    expect(response.text).toContain("value=\"yes\" checked");
```
Comment (markpit) [inline test/controller/objecting.entity.name.controller.spec.unit.ts:91]: could you add a not to contain 'no' and similar in the other test for 'yes'

Comment (mattch1) [reply test/controller/objecting.entity.name.controller.spec.unit.ts:91]: Already done.

Comment (ICoxCH) [reply test/controller/objecting.entity.name.controller.spec.unit.ts:91]: I think Mark means adding
`expect(response.text).not.toContain("value=\"no\" checked");`
In this test, and the same for 'yes' in the test below.
Checking that it is just "Yes" checked here and "No" is not checked

Comment (markpit) [reply test/controller/objecting.entity.name.controller.spec.unit.ts:91]: yes , we are checking if yes is checked, but the test will pass if both yes and no are checked

Comment (mattch1) [reply test/controller/objecting.entity.name.controller.spec.unit.ts:91]: I did know you could check both I'll have to try that out - updated the tests

**Review by markpit — APPROVED** [review]

**Review by ICoxCH — APPROVED** [review]

**Pull request merged**